app/controllers/admin/contacts_controller.rb

```diff
@@ -47,12 +48,14 @@ def create
     end
 
     def destroy
-      contact = Contact.find(params[:id])
-      location = contact.location
+      location = Location.find(params[:location_id])
 
       authorize location
 
-      contact.destroy
+      resource = location.resource_contacts.find_by(contact_id: params[:id])
+      contact = resource.contact
+      resource.destroy
+
       redirect_to admin_location_url(location),
                   notice: "Contact '#{contact.name}' was successfully deleted."
```

Comment:
`was removed from #{location.name}` might be more correct here as a message. Thinking aloud, not a needed change.